src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx

```diff
@@ -3,223 +3,218 @@
  * See the LICENSE file in the repository root folder for details.
  */
 
-import { createContext, useState, SyntheticEvent } from 'react';
-import { Container, Form } from 'react-bootstrap';
-import { WbsNumber, WorkPackage, WbsElementStatus } from 'shared';
-import { useAuth } from '../../../hooks/auth.hooks';
-import { useAllUsers } from '../../../hooks/users.hooks';
-import { useEditWorkPackage } from '../../../hooks/work-packages.hooks';
-import { routes } from '../../../utils/Routes';
+import { DescriptionBullet, validateWBS, WorkPackage } from 'shared';
 import { wbsPipe } from '../../../utils/Pipes';
-import EditableTextInputList from '../../../components/EditableTextInputList';
+import { routes } from '../../../utils/Routes';
+import { useAllUsers } from '../../../hooks/users.hooks';
+import { useAuth } from '../../../hooks/auth.hooks';
+import PageTitle from '../../../layouts/PageTitle/PageTitle';
+import PageBlock from '../../../layouts/PageBlock';
 import ErrorPage from '../../ErrorPage';
 import LoadingIndicator from '../../../components/LoadingIndicator';
-import PageBlock from '../../../layouts/PageBlock';
-import PageTitle from '../../../layouts/PageTitle/PageTitle';
-import { EditableTextInputListUtils } from '../../CreateWorkPackagePage/CreateWPForm';
-import DependenciesList from '../WorkPackageViewContainer/DependenciesList';
-import EditModeOptions from './EditModeOptions';
-import WorkPackageEditDetails from './WorkPackageEditDetails';
 import { useQuery } from '../../../hooks/utils.hooks';
+import { useFieldArray, useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import { Button, Box, TextField, Grid, IconButton } from '@mui/material';
+import ReactHookTextField from '../../../components/ReactHookTextField';
+import ReactHookEditableList from '../../../components/ReactHookEditableList';
+import { useEditWorkPackage } from '../../../hooks/work-packages.hooks';
+import WorkPackageEditDetails from './WorkPackageEditDetails';
+import DeleteIcon from '@mui/icons-material/Delete';
+
+/*
+ * maps a description bullet list to the object needed for forms
+ * can't use `id` instead of `bulletId` because react-hook-forms uses id built in for arrays of objects
+ */
+const bulletsToObject = (bullets: DescriptionBullet[]) =>
```

Comment:
if we're using the exact same stuff in this and project edit we should probably abstract them? if we do though we need to document exactly the input and output and why so that it's not too confusing lol because the use case is so specific